packages/pharos/src/components/coach-mark/pharos-coach-mark.ts

```diff
@@ -0,0 +1,149 @@
+import { PharosElement } from '../base/pharos-element';
+import { PharosButton } from '../button/pharos-button';
+import { PharosHeading } from '../heading/pharos-heading';
+import { html } from 'lit';
+import { property } from 'lit/decorators.js';
+import type { TemplateResult, CSSResultArray } from 'lit';
+import { coachMarkStyles } from './pharos-coach-mark.css';
+import ScopedRegistryMixin from '../../utils/mixins/scoped-registry';
+import { autoUpdate, computePosition, flip, offset } from '@floating-ui/dom';
+
+export type Side = 'top' | 'right' | 'bottom' | 'left';
+export type Alignment = 'start' | 'center' | 'end';
+export type Delay = 'none' | 'short' | 'long';
+export type Variant = 'light' | 'dark';
+
+/**
+ * Pharos coach mark component.
+ *
+ * @tag pharos-coach-mark
+ *
+ */
+export class PharosCoachMark extends ScopedRegistryMixin(PharosElement) {
+  static elementDefinitions = {
+    'pharos-button': PharosButton,
+    'pharos-heading': PharosHeading,
+  };
+
+  public static override get styles(): CSSResultArray {
+    return [coachMarkStyles];
+  }
+
+  /**
+   * Indicates that the coach mark should not be displayed on the page
+   * @attr hide
+   */
+  @property({ type: Boolean, reflect: true })
+  public hide = true;
+
+  /**
+   * Indicates which side of the base element the coachmark should appear on
+   * @attr side
+   * @type {Side}
+   */
+  @property({ reflect: true })
+  public side: Side = 'bottom';
+
+  /**
+   * Indicates how the coach mark carat should be aligned in relation to the coach mark content
+   * @attr alignment
+   * @type {Alignment}
+   */
+  @property({ reflect: true })
+  public alignment: Alignment = 'center';
+
+  /**
+   * Dictates how long to wait between coach mark trigger and start of coach mark fade in animation
+   * @attr delay
+   * @type {Delay}
+   */
+  @property({ reflect: true })
+  public delay: Delay = 'short';
+
+  /**
+   * Text content for the coach mark header
+   * @attr header
+   * @type {String}
+   */
+  @property({ reflect: true })
+  public header = '';
+
+  /**
+   * Style variant
+   * @attr variant
+   * @type {Variant}
+   */
+  @property({ reflect: true })
+  public variant: Variant = 'dark';
+
+  /**
+   * Set minimum width of coach mark
+   * @attr width
+   * @type {string}
+   */
+  @property({ reflect: true })
+  public width = '30ch';
+
+  private computedSide = 'bottom';
```

Comment:
Our convention is typically to prefix private members with underscore.

Reply:
addressed in [497a642](https://github.com/ithaka/pharos/pull/608/commits/497a6429f8859b2f01eb4f83d8449c05977817da)